docs/framework/docker/aspnetmvc.md

```diff
@@ -0,0 +1,240 @@
+assets in one directory tree that you will copy to your target image.
+Right-click on your solution in Visual Studio, and select "Publish".
+Click the "Custom" profile button, and then select "File System" as the
+method (see image). Choose a new directory in your solution folder. (The
+downloaded sample uses "containerImage" as the folder name).
+
+![Publish Connection][publish-connection]
+
+Next, open the "Advanced" section of the "File Publish" options. Select
+"Precompile during publishing". This optimization means that you be
+compiling views in the Docker container, you are copying the precompiled
+views.
+
+![Publish Settings][publish-settings]
+
+Click publish, and Visual Studio will copy all the needed assets to the
+destination folder. 
+
+## Build the image
+
+To run your application, you'll build an image based on the `microsft/aspnet`
+image located on [Docker Hub](https://hub.docker.com/r/microsoft/aspnet/).
+
+You'll compose this image using the `microsoft/aspnet` image as the base:
+
+```
+FROM microsoft/aspnet
+```
+
+The next command creates the directory that holds your application:
+
+```
+RUN mkdir C:\randomanswers
+```
+
+After that, you run a powershell command in the Docker image to setup
+the new site: 
+
+```
+RUN powershell -NoProfile -Command \
+    Import-module IISAdministration; \
+    New-IISSite -Name "ASPNET" -PhysicalPath C:\randomanswers -BindingInformation "*:8000:"
+```
+
+Following that, you instruct your Docker image to listen on port 8000: 
+
+```
+EXPOSE 8000
+```
+
+And last, you copy everything from your publish directory to the Docker image:
+
+```
+ADD containerImage/ /randomanswers
+```
+
+Run a Docker build command to create the image that will run your asp.net
+application. Open a powershell window, and type the following command in
+the solution directory:
+
+```
+docker build -t mvcrandomanswers .
+```
+
+Once that command completes, you can run the `docker images` command
+to see information on the new image:
+
+```
+REPOSITORY                    TAG                 IMAGE ID            CREATED             SIZE
+mvcrandomanswers              latest              86838648aab6        2 minutes ago       8.104 GB
+```
+
+The IMAGE ID will be different on your machine. Now, let's run the applicaton.
+
+## Start a container
+
+You start a container by executing the `docker run` command:
+
+```
+docker run -d -p 8000:8000 --name randomanswers mvcrandomanswers
+```
+
+The `-d` argument tells Docker to start the image in detached mode. That
+means the Docker image runs disconnected from the current shell.
+
+The `-p 8000:8000` argument tells docker how to map incoming ports. In this
+example, I'm using port 8000 on both the host and the container.
+
+The `--name randomanswers` gives a name to the running container. You can use
+this name instead of the container ID in most commands.
+
+The `mvcrandomanswers` is the name of the image to start.
+
+## Find the IP address
```

Comment:
It currently works in Linux to just do localhost:<mapped port> from the container host. That's why it's really confusing